gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
+            rplanet=None,
+            pressure_0=None,
+            co_ratio=0.55,
+            feh_ratio=0.0,
+            mass_frac=None,
+    ):
+        """
+        Calculate the transit spectrum. This function avarages T-p profiles in
+        the terminator region and uses poorman equilbriums chemistry.
+
+        Parameters
+        ----------
+        mmw: float or 1D-array
+            Mean molecular weight (in atomic units). Will be globally uniform if
+            float or horizonatally uniform if 1D.
+        lat_points : int
+            Number of latitude points considered per terminator.
+        lon_resolution : float
+            Longitude angle to avarage over at the equator of the terminators. In !degrees!.
+        gravity: float
+            surface gravity in !cgs!. Will default to the value provided by GCMT.
+        rplanet: float
+            planet radius in !cm!. Will default to the value provided by GCMT.
+        pressure_0: float
+            pressure level of the gravity and radius. Will default to the value provided by GCMT.
+        co_ratio: float, optional
+            The C/O ratio. Currently only one global value allowed. Defaults to 0.55.
+        feh_ratio: float, optional
+            The metalicity ratio. Currently only one global value allowed.
+            Defaults to 0.0.
+        mass_frac: List[List[Dict]], optional
+            The mass fractions for each t_p point. Structure is as follows: mass_frac[i][j]['key']
+            - i: [2 elements] 0 for morning terminator, 1 for evening terminator
+            - j: [lat_points elements] latitude point starting from lowest (closeset to lat = -90) to highest (closest to lat = 90)
+            - dict must contain all mass fractions  elements given as opacity species to prt
+
+
+        Returns
+        -------
+        spectra: xr.DataArray
+            Dataarray containing the spectrum.
+            The spectrum is normed to the stellar spectrum.
+
+        """
+        from petitRADTRANS.poor_mans_nonequ_chem import interpol_abundances
+        from petitRADTRANS import nat_cst as nc
+
+        # prepare chemistry input (make them 1D arrays)
+        pres = self.dsi[c["Z"]].values
+        co_ratios = np.ones_like(self.dsi[c["Z"]].values) * co_ratio
+        feh_ratios = np.ones_like(self.dsi[c["Z"]].values) * feh_ratio
+        tlist = [('H2O', 'H2O_HITEMP'),
+                 ('H2O', 'H2O_Exomol_R_1'),
+                 ('CO', 'CO_all_iso_HITEMP'),
+                 ('Na', 'Na_allard'),
+                 ('K', 'K_allard')]
+
+        # check if gravity is given, if not use gcm value
+        if gravity is None:
+            gravity = self.dsi.attrs.get(c["g"]) * 100
+
+        # check if radius is given, if not use gcm value
+        if rplanet is None:
+            rplanet = self.dsi.attrs.get(c["R_p"])
+
+        # check if pressure is given, if not use gcm value
+        if pressure_0 is None:
+            pressure_0 = np.max(pres)
```

Comment:
careful with units! 
But you can use `np.max(self.prt.press)*1e-6`, to be on the safe side

Reply:
Pressure can be Pa and bar internally? If yes I will add case handling

Reply:
Just use prt.press. That one is in cgs. Always.